src/algorithms/FinField.jl

```diff
@@ -12,55 +12,66 @@
 
 struct FinFieldIterator{F}
    basis::Vector{F}
-   current::Vector{Int}
+   coeffs::Vector{Int64} # coefficients in the base field, assumed to be a prime field;
+                         # even for characteristic > typemax(Int64), it's impossible to
+                         # exhaust that many scalars with iteration
 
    function FinFieldIterator(f)
       deg = degree(f)
+      characteristic(f)^deg == order(f) ||
```

Comment:
```suggestion
      BigInt(characteristic(f))^deg == order(f) ||
```
Otherwise this might overflow, right?

Reply:
I hesitated, but for Nemo's `FqNmodFiniteField` and `FqFiniteField`, the characteristic is already an `fmpz` so it won't overflow. I would assume that In general, it's a good idea to have`characteristic(f)` be an integer type big enough to represent `order(f)`. And in the worst case, this check will probably fail and someone will open an issue; but I guess it's a no-brainer to just use `BigInt` indeed.